depinject/container.go

```diff
@@ -130,7 +121,26 @@ func (c *container) getResolver(typ reflect.Type, key string) (resolver, error)
 		c.resolvers[mapType] = &mapOfOnePerModuleResolver{r}
 	}
 
-	return c.resolvers[typ], nil
+	res := c.resolvers[typ]
+
+	if res == nil && typ.Kind() == reflect.Interface {
+		var found bool
+		for k, r := range c.resolvers {
+			if k.Implements(typ) {
+				c.logf("Found candidate resolver, implicitly binding interface %v to implementing type %v.", typ, k)
+				if found {
+					// TODO provide example YAML binding when the implementation is done
+					msg := fmt.Sprintf("<stub> Multiple implementations found for interface %v.  Specify an explicit type binding in app.yaml", typ)
```

Comment:
depinject doesn't know about yaml. We should return a typed error here and let appconfig translate it to something relating to the config